subprojects/docs/src/docs/userguide/dependencyLocking.adoc

```diff
@@ -0,0 +1,144 @@
+Because for the latter, the _GAV_ remains the same, it is the content - metadata and/or artifact - that changes.
+====
+
+=== Enabling locking on configurations
+
+Locking of a configuration happens through the `ResolutionStrategy`:
+
+++++
+<sample id="locking-one" dir="userguide/dependencyManagement/dependencyLocking/lockingSingleConfiguration" title="Locking a specific configuration">
+    <sourcefile file="build.gradle" snippet="locking-one"/>
+</sample>
+++++
+
+Or the following, as a way to lock all configurations:
+
+++++
+<sample id="locking-all" dir="userguide/dependencyManagement/dependencyLocking/lockingAllConfigurations" title="Locking all configurations">
+    <sourcefile file="build.gradle" snippet="locking-all"/>
+</sample>
+++++
+
+[NOTE]
+====
+Only configurations that can be resolved will have a lock file attached to them. Applying locking on non resolvable-configurations is simply a no-op.
+====
+
+=== Lock file location and format
+
+Lockfiles will be located in the folder `gradle/dependency-locks` inside the project or subproject directory.
+Each file is named by the configuration it locks and has the `lockfile` extension.
+
+The content of the file is a module notation per line, with a header giving some context.
+Module notations are ordered alphabetically, to ease diffs.
+
+++++
+<sample id="locking-all" dir="userguide/dependencyManagement/dependencyLocking/lockingSingleConfiguration" title="Lockfile content">
+    <sourcefile file="gradle/compileClasspath.lockfile"/>
+</sample>
+++++
+
+=== Running a build with a lockfile present
+
+The moment a build needs to resolve a configuration that has locking enabled and it finds a matching lockfile,
+it will use it to verify that the given configuration still resolves the same versions.
+The validation is as follows:
+
+* Existing entries in the lockfile must be matched in the build
+** A version mismatch or missing resolved module causes a build failure
+* Resolution result must not contain extra dependencies compared to the lock file content
+
+=== Generating and updating lockfiles
+
+In order to generate or update a lockfile, you specify the `--write-locks` command line argument in addition to the normal tasks that would trigger configurations to be resolved.
+This will cause the creation of a lockfile for each resolved configuration in that build execution.
+Note that if a lockfile existed previously, it is overwritten.
+
+==== Lock all configurations in one build execution
+
+When locking multiple configurations, you may want to lock them all at once, during a single build execution.
+
+For this, you have two options:
+
+* Run `./gradlew dependencies --write-locks`.
+This will effectively lock all resolvable configurations that have locking enabled.
+Note that in a multi project setup, `dependencies` only is executed on _one_ project, the root one in this case.
+* Declare a custom task that will resolve all configurations
```

Comment:
❓ Do we even need to document this 2nd option? Why would I want to do this instead?

Reply:
As you may need to resolve only specific configurations. For example in the native world, you may not be able to resolve all configurations at once as they might require to be resolved on different platforms.

Reply:
Added a word to the doc about that example.